src/query/storage/validator/storage.go

```diff
@@ -115,17 +116,23 @@ func PromResultToSeriesList(promReadResp prometheus.PromResp, tagOptions models.
 			}
 		}
 
+		metricName := string(tagOptions.MetricName())
 		tags := models.NewTags(len(result.Metric), tagOptions)
 		for name, val := range result.Metric {
-			tags = tags.AddTag(models.Tag{
-				Name:  []byte(name),
-				Value: []byte(val),
-			})
+			if name == metricName {
+				tags = tags.SetName([]byte(val))
+			} else {
+				tags = tags.AddTag(models.Tag{
+					Name:  []byte(name),
+					Value: []byte(val),
+				})
+			}
 		}
 
 		name, exists := tags.Name()
 		if !exists {
-			return nil, errors.New("metric name does not exist")
+			// return nil, errors.New("metric name does not exist at all ever")
```

Comment:
Hm, don't we want to restore this return err?  What possibility would we want to allow for empty series names?

Reply:
Added a comment explaining why it's fine to set to `default` 👍

Reply:
Ah interesting.  Thx for the detailed response.